internal/client.go

```diff
@@ -713,6 +723,20 @@ type (
 		callbacks []*commonpb.Callback
 	}
 
+	// StartWorkflowOperation is a type of operation that can be executed as part of a workflow start.
+	StartWorkflowOperation interface {
+		isStartWorkflowOperation()
+	}
```

Comment:
One gripe I have with this name is that it makes it seem like it's the "StartWorkflow" operation (ie an operation to start a workflow).

Reply:
Additional works for me, or `WithStartWorkflowOperation` which mimics the existing signal with start names.

Reply:
`WithStartWorkflowOperation` and `StartWorkflowAdditionalOperation` both fine with me.

I like "With Start" as it mirrors how we talk about it in the docs and conceptual guides.

Reply:
The issue is the name of the update operation is unrelated to this use, it's named generically as if it's used in many places